### PR TITLE
Putusers

### PR DESCRIPTION
**Pull request by Sleepy-girl**

### CODE REVIEW

File: backend/src/api/users/users.controllers.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+const UserDB = require('./user.model');
+const catchAsync = require('../../utils.js/catchAsync');
+
+const updateUsersController = catchAsync(async (req, res, next) => {
+  if (req.user && req.body) {
```
Comment (alucardpetrovich) [inline backend/src/api/users/users.controllers.js]: you would always have req.user if authorize middleware passed execution to this middleware

Comment (alucardpetrovich) [inline backend/src/api/users/users.controllers.js]: if req.body is not enough here. You haven't done any validation

File: backend/src/api/users/users.router.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+const { Router } = require('express');
```
Comment (alucardpetrovich) [inline backend/src/api/users/users.router.js]: Also where are auto-tests? It's one of acceptance criterias

File: backend/src/api/users/users.middleware.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+const UserSchema = Joi.object({
+  balance: Joi.number(),
+  flatPrice: Joi.number(),
+  flatSquareMeters: Joi.number(),
+  totalSalary: Joi.number(),
+  passiveIncome: Joi.number(),
+  incomePercentageToSavings: Joi.number(),
+});
+
+const validationMiddleware = schema => async (req, res, next) => {
```
Comment (alucardpetrovich) [inline backend/src/api/users/users.middleware.js]: There is already function for validation in utils folder

File: backend/src/api/users/users.controllers.js (hunk 1 above)
Comment (alucardpetrovich) [inline backend/src/api/users/users.controllers.js]: I've meant to remove this `if` completely

Comment (Sleepy-girl) [reply backend/src/api/users/users.controllers.js]: done)

File: backend/src/api/users/users.router.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+const { updateUsersController } = require('./users.controllers');
+const usersRouter = Router();
+const { authorize } = require('../../utils.js/authMiddleware');
+const { validate } = require('../../utils.js/validate');
+const Joi = require('joi');
+
+const UserSchema = Joi.object({
+  balance: Joi.number(),
```
Comment (alucardpetrovich) [inline backend/src/api/users/users.router.js]: all fields should be required in this schema

File: backend/src/api/users/users.controllers.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+const UserDB = require('./user.model');
+const catchAsync = require('../../utils.js/catchAsync');
+
+const updateUsersController = catchAsync(async (req, res, next) => {
+  if (req.body) {
```
Comment (alucardpetrovich) [inline backend/src/api/users/users.controllers.js]: i still see this if ))

**Pull request merged**